python-spec/src/somacore/data.py

```diff
@@ -215,7 +317,37 @@ def read(
             some_dense_array.read(...).csrs().all()
             # -> a single flattened sparse CSR matrix
 
-        TODO: Additional per-param documentation.
+        :param coords: A per-dimension sequence of coordinates defining
+            the range to be read.
+        :param batch_size: The size of batches that should be returned
+            from a read. See :class:`options.BatchSize` for details.
+        :param partitions: Specifies that this is part of a partitioned read,
+            and which partition to include, if present.
+        :param result_order: The order to return the results in.
```

Comment:
give example values; mysterious otherwise

Reply:
Linked to `ResultOrder`